python/ray/ml/config.py

```diff
@@ -105,6 +115,47 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         bundles = trainer_bundle + worker_bundles
         return PlacementGroupFactory(bundles, strategy=self.placement_strategy)
 
+    @classmethod
+    def validate_config(
+        cls,
+        config: Union[ScalingConfig, "ScalingConfigDataClass"],
+        allowed_keys: Iterable[str],
+    ):
+        """
+        Validate config by raising an exception if any of allowed_keys
+        differs from the default value.
+
+        Args:
+            config: Config to check.
+            allowed_keys: ScalingConfigDataClass attribute keys that
+                can have a value different than the default one.
+        """
+        default_config = ScalingConfigDataClass()
+        if not isinstance(config, ScalingConfigDataClass):
+            config = ScalingConfigDataClass(**config)
+
+        allowed_keys = set(allowed_keys)
+
+        bad_allowed_keys = [key for key in allowed_keys if key not in config.__dict__]
+        if bad_allowed_keys:
+            raise KeyError(
+                f"Allowed key(s) {bad_allowed_keys} are not valid. "
+                f"Valid keys: {list(config.__dict__.keys())}"
+            )
+
+        prohibited_keys = set(default_config.__dict__) - allowed_keys
```

Comment:
I think I had this comment on the other PR as well.
why would a prohibited_key live on default_config dict?

Reply:
We use this method to restrict the user to a subset of keys (allowed_keys). Therefore, all the keys that are not allowed are prohibited from being set by the user.

Reply:
Added two comments to make this clearer